examples/rust/providers/messaging-nats/src/connection.rs

```diff
@@ -0,0 +1,67 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+const DEFAULT_NATS_URI: &str = "0.0.0.0:4222";
+
+const CONFIG_NATS_SUBSCRIPTION: &str = "subscriptions";
+const CONFIG_NATS_URI: &str = "uri";
+
+/// Configuration for connecting a NATS client.
+/// More options are available if you use the json than variables in the values string map.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct ConnectionConfig {
+    /// List of topics to subscribe to for a given component
+    #[serde(default)]
+    pub subscriptions: Vec<String>,
+
+    /// URI for the NATS cluster to connect to
+    #[serde(default)]
+    pub uri: String,
+}
+
+impl ConnectionConfig {
+    /// Merge a given [`ConnectionConfig`] with another, coalescing fields and overriding
+    /// where necessary
+    pub fn merge(&self, extra: &ConnectionConfig) -> ConnectionConfig {
+        let mut out = self.clone();
+        if !extra.subscriptions.is_empty() {
+            out.subscriptions = extra.subscriptions.clone();
+        }
+        // If the default configuration has a URL in it, and then the link definition
+        // also provides a URL, the assumption is to replace/override rather than combine
+        // the two into a potentially incompatible set of URIs
+        if !extra.uri.is_empty() {
+            out.uri= extra.uri.clone();
+        }
+        out
+    }
+}
+
+impl Default for ConnectionConfig {
+    fn default() -> ConnectionConfig {
+        ConnectionConfig {
+            subscriptions: vec![],
+            uri: DEFAULT_NATS_URI.to_string(),
+        }
+    }
+}
+
+impl ConnectionConfig {
+    /// Construct configuration Struct from the passed config values
+    pub fn from_map(values: &HashMap<String, String>) -> Result<ConnectionConfig> {
```

Comment:
Nit: This should probably just be a `From` implementation for `ConnectionConfig` (also would prefer that it takes owned data instead of borrowed, but that is more of a super nit)

Reply:
Oh, and this definitely doesn't need to be a `Result` as there are no error cases

Reply:
Yep good call. Leftover from the "real" impl